app/models/comment.rb

```diff
@@ -5,8 +5,8 @@ class Comment < ActiveRecord::Base
     :subject, :hostname, :comment,
     :status, :format, :thread, :timestamp
 
-  belongs_to :node, foreign_key: 'nid', touch: true,
-                    dependent: :destroy, counter_cache: true
+  belongs_to :node, foreign_key: 'nid', touch: true, counter_cache: true
```

Comment:
Hmm, does this have to do with the comment changes? What was the issue? I wasn't sure `counter_cache` was working and would like to demonstrate this with a test if that's OK... a unit test?

Reply:
Hi @jywarren. Actually I haven't removed counter_cache but 'dependent: :destroy'.
Actually the error was occurring because some comments did not belong to an answer or a node. Perhaps there is no need of dependent: :destroy here but the node model (has_many comments, dependent: :destroy). Just realized that dependent destroy also needs to be added to answer model here:-
https://github.com/publiclab/plots2/blob/b978d2bd8912a3c4cbffd3a320f44f05deb8764d/app/models/answer.rb#L9
Please correct me if I am wrong!

Reply:
Hi @jywarren , The tests are written. Please have a look